scan/lib/scan/test_command_generator.rb

```diff
@@ -31,8 +31,9 @@ def options
 
         options = []
         options += project_path_array
-        options << "-sdk '#{config[:sdk]}'" if config[:sdk]
-        options << destination # generated in `detect_values`
+        options << "-sdk '#{config[:sdk]}'" if config[:sdk] && !config[:build_for_testing]
+        options << destination unless config[:build_for_testing] # generated in `detect_values`
+        options << "  -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO" if config[:build_for_testing]
```

Comment:
Just to clarify, there's the use case of "I want to run the same testing target on loads of different devices" that I think you are alluding to. But there's also the use case of "I want to do all my compiling for my multiple testing targets on one machine and reap the speedup reward of not recompiling shared dependencies, but then actually run the tests on different machines". I'm interested in the latter.